### ISSUE TITLE
nullability of value parameter in properties

### ISSUE THREAD

**Issue by dlech** [dotnet/roslyn#38943]
**Version Used**: master (23 September 2019) via https://sharplab.io

**Steps to Reproduce**:

```C#
#nullable enable

using System;
using System.Diagnostics.CodeAnalysis;

public class C {
    string? Getter() {
        return null;
    }
    
    void Setter(string s) {
    }
    
    [DisallowNull]
    public string? Property {
        get => Getter();
        set => Setter(value); // get unexpected warning here
    }
    
    public void Test() {
        Property = ""; // no warning here as expected
        Property = null; // get warning here as expected
        Setter(Property); // get warning here as expected
    }
}
```

**Expected Behavior**: Given a property with the `DisallowNull` attribute it should mean that the return value can be null but `value` parameter of the setter cannot be null.

**Actual Behavior**: The code analysis gets this right when assigning to a property, but not in the property declaration.

**Comment by gafter** [comment]
I believe this is by design.  It was not intended that `[DisallowNull]` have an effect on the type of the parameter inside the method.

**Comment by dlech** [comment]
This gives an unexpected warning as well

```c#
#nullable enable

using System;
using System.Diagnostics.CodeAnalysis;

public class C {
    string? Getter() {
        return null;
    }
    
    void Setter(string s) {
    }
    
    [MaybeNull]
    public string Property {
        get => Getter(); // get unexpected warning here
        set => Setter(value);
    }
    
    public void Test() {
        Property = ""; // no warning here as expected
        Property = null; // get warning here as expected
        Setter(Property); // get warning here as expected
    }
}
```

So it doesn't seem possible to have a property with different nullability for the getter and setter when the getter and setter are expression bodied.
